### PR TITLE
Formalize HR agent lifecycle

### LINKED ISSUE #9: Formalize agent lifecycle and HR actions
Turn HR role creation from a single escape-code side effect into an explicit lifecycle model for agents in the organization.

Acceptance criteria:
- Define lifecycle states such as proposed, active, paused, and retired.
- Make role creation validate name, description, max organization size, and duplicate handling.
- Record who requested or approved lifecycle changes in the session transcript or memory layer.
- Add tests for create, duplicate rejection, capacity limit, pause, and retirement behavior.
- Keep existing create_role behavior compatible during migration.

### PR DESCRIPTION
**Pull request by displague**
## Summary
- add explicit in-memory lifecycle state and lifecycle events for roles
- preserve `org.create_role` compatibility while adding validation, requester, and approver tracking
- add trusted `org.pause_role` and `org.retire_role` lifecycle tools with transition validation
- record lifecycle tool output in session system events and document the runtime behavior

## Validation
- `python -m unittest` (52 tests)
- `python <skill-creator>/scripts/quick_validate.py .github/skills/robits-runtime`
- `git diff --check`

Acting as Codex GPT-5.5 on behalf of displague.

Closes #9

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -237,6 +244,166 @@ def as_chat_completion_tools(self):
...
+
+@dataclass
+class LifecycleEvent:
+    action: str
+    agent_name: str
+    lifecycle_state: str
+    requested_by: str | None = None
+    approved_by: str | None = None
+    reason: str | None = None
+    created_at: str = field(default_factory=lambda: datetime.now().isoformat(timespec="seconds"))
```
[the remark this answers is not in the document]
Comment (displague) [reply main.py]: Addressed in 40dbb8c by recording lifecycle event timestamps with timezone-aware UTC. Added a regression assertion that lifecycle event timestamps include the UTC offset.

Acting as Codex GPT-5.5 on behalf of displague.

File: main.py [hunk 2]
```diff
@@ -237,6 +244,166 @@ def as_chat_completion_tools(self):
...
+    if lifecycle_state not in LIFECYCLE_STATES:
+        return f"Error: Invalid lifecycle state '{lifecycle_state}'."
+    if normalized_name not in employee_dict:
+        return f"Error: Role '{normalized_name}' not found."
+
+    role = employee_dict[normalized_name]
+    record_lifecycle_event(
+        role,
+        action=action,
+        lifecycle_state=lifecycle_state,
```
[the remark this answers is not in the document]
Comment (displague) [reply main.py:372]: Addressed in 40dbb8c by adding transition validation: pause only from active, retire from active or paused, and invalid transitions return an error without appending a lifecycle event. Added regression coverage.

Acting as Codex GPT-5.5 on behalf of displague.

File: tools.yaml [hunk 3]
```diff
@@ -11,20 +11,74 @@
...
-        return f"Error: Role '{role_name}' already exists."
+    return create_lifecycle_role(
+        employee_dict,
+        role_name,
+        role_description,
+        requested_by=requested_by,
+        approved_by=approved_by,
+    )
+- name: org.pause_role
+  description: Pause an active role in the organization.
```
[the remark this answers is not in the document]
Comment (displague) [reply tools.yaml:32]: Addressed in 40dbb8c by enforcing the active-only pause constraint in the implementation, so the tool description now matches behavior.

Acting as Codex GPT-5.5 on behalf of displague.

File: README.md [hunk 4]
```diff
@@ -33,6 +33,11 @@ The simulation runs in a session, where organization members communicate through
 
 The runtime can parse JSON tool instructions, load trusted tools from `tools.yaml`, and execute registered tools by name. Tools use namespaced identifiers such as `org.create_role`, with short aliases only where compatibility is useful.
 
+HR lifecycle tools keep role creation compatible while adding explicit states:
+`active`, `paused`, and `retired` roles record lifecycle events with optional
+requester, approver, and reason fields. `org.create_role` creates an active role;
+`org.pause_role` and `org.retire_role` change an existing role's lifecycle state.
```
[the remark this answers is not in the document]
Comment (displague) [reply README.md]: Addressed in 40dbb8c by documenting `proposed` alongside the other lifecycle states and clarifying that current trusted tools create active roles directly.

Acting as Codex GPT-5.5 on behalf of displague.
